tests/test_utils.py

```diff
@@ -550,3 +553,34 @@ def clean_athena_db(connection_string: str, db_name: str, table_to_keep: str) ->
     finally:
         athena_connection.close()
         engine.dispose()
+
+
+def retry_probabilistic_test(
+    max_num_retries: int = MAX_PROBABILISTIC_TEST_ASSERTION_RETRIES,
+) -> Callable:
+    def execute_probabilistic_test_decorator(func: Callable) -> Callable:
+        @wraps(func)
+        def run_pytest_method(*args, **kwargs) -> Any:
+            assertion_error: Optional[AssertionError] = None
+
+            all_assertions_passed: bool = False
+
+            idx: int = 0
+            while idx < max_num_retries:
+                try:
+                    func(*args, **kwargs)
+                    all_assertions_passed = True
+                except AssertionError as assertion_error:
```

Comment:
Maybe use something more broad here like Exception? What if we aren't asserting in a test?

Reply:
@cdkini Why would we not assert in a test?

Reply:
hmm maybe not super present here but we have tests on the Cloud backend that don't assert and use the lack of errors to signify tests passing.

I think the idea still stands though - what if our test fails before we get to our asserts?

Reply:
@cdkini Then that kind of failure has to be handled differently, because this retry mechanism is specifically for tests that may have intermittent failures, due to the mechanism of parameter estimation being probabilistic; these failures would NOT be due to other errors (e.g., a threading issue).